examples/apps/generic_client/generic_client_interface.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+
+
+class GenericClientInterface(ABC):
+    """
+    Generic client interface
+    """
+    def __init__(self):
+        super().__init__()
+
+    @abstractmethod
+    def get_first_active_worker_id(self):
+        """
+        Get first active work order id
+        returns worker id
+        """
+        pass
+
+    @abstractmethod
+    def get_worker_details(self, worker_id):
+        """
+        Fetch worker details for given worker id
+        @param worker_id is identifier of worker
+        returns worker details object
+        """
+        pass
+
+    @abstractmethod
+    def do_worker_verification(self, worker_obj):
+        """
+        Do worker verification on proof data if it exists
```

Comment:
worker verification involves verifying all relevant fields in worker details.  It also verifies other worker details fields like encryptionKeySignature. @manju956 had added encryptionKeySignature verification of worker details in one of the PR.

Reply:
It can be taken separate PR as i am going to introduce the MR enclave verification.